### DISCUSSION TITLE
Issue setting default value for struct

### DISCUSSION THREAD

**Discussion by brk3** [cue-lang/cue#1794] [Q&A] [answered]
I'd like to have a struct set to a default value, while allowing the value to be overridden if required. My issue is that the default doesn't seem to work for structs?

input:
```
#Person: {
  info: [string]: string | *{
    name:     "paul"
    location: "dublin"
  }
}

// Why isn't the default showing?
Paul: #Person

// Override works as expected
John: #Person & {
  info: {
    name: "john"
  }
}
```

output:
```
$ cue eval test.cue
#Person: {
    info: {}
}
Paul: {
    info: {}
}
John: {
    info: {
        name: "john"
    }
}

```

Can anyone point me in the right direction here?

**Comment by eonpatapon** [comment]
Looks like your `#Person` definition should be:

```cue
#Person: {
  info: string | *{
    name:     "paul"
    location: "dublin"
  }
}
```

To achieve what you want.

Here you have `info: [string]: someValue`. If no field is defined in `info` you won't get any value. For any `string` field (so any field) defined in `info` you will get `someValue`.

```
John: #Person & {
  info: {
    name: "john"
  }
}
```

Works because it matches the case: `info: [string]: string`

**Reply by brk3** [reply]
Thanks for the response. The issue with your approach (I think) is that I can't then override `info` with a struct:

```
John.info: conflicting values string and {name:"john"} (mismatched types string and struct):
```

If I don't take the default then I have to provide a `string` as a value for `info` which is not what I want: https://cuelang.org/play/?id=4JtTXvmrmpN#cue@export@cue

**Comment by brk3** [comment, marked as answer]
@eonpatapon suggested another answer on Slack that works perfectly:

```
package main

#Person: {
	_info: [string]: string

	info: *{
		name:     "paul"
		location: "dublin"
	} | _info

}

Paul: #Person

John: #Person & {
	info: {
		name: "john"
		foo:  "bar"
	}
}
```

**Reply by myitcv** [reply]
The hidden field is actually not required. Only some braces around `{[string]:string}`:

https://cuelang.org/play/?id=pPAU2k1Hw5_l#cue@export@cue

**Reply by brk3** [reply]
Even better, I had a feeling there should be way like this also. Thanks very much @myitcv
